src/test/java/net/masterthought/cucumber/util/CounterTest.java

```diff
@@ -0,0 +1,15 @@
+package net.masterthought.cucumber.util;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertNotEquals;
+
+public class CounterTest {
+    @Test
+    public void shouldIncrement() {
```

Comment:
Next returns the next value, not the current.
The initial value, 0, can only be seen if `intValue` is used.
In fact, to simplify, I'm tempted to make `Counter` contain `MutableInt`, not inherit from it, and only have one API: `next`.